File: web/src/main/java/com/example/reservation/controller/ReservationController.java
package com.example.reservation.controller;

import com.example.reservation.dto.ReservationRequest;
import com.example.reservation.service.ReservationService;
import com.example.response.BaseResponse;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class ReservationController {
    private final ReservationService reservationService;

    @PostMapping("/v1/reservation")
    public BaseResponse getMoviesNowShowing(@RequestBody @Valid ReservationRequest reservationRequest) throws InterruptedException {
        reservationService.reserveSeat(reservationRequest);
        return new BaseResponse<>();
    }
}

File: movie/api/src/main/kotlin/com/example/movie/controller/ReservationController.kt
package com.example.movie.controller

import com.example.movie.dto.ReservationResponse
import com.example.movie.request.ReservationRequest
import com.example.movie.service.ReservationUseCase
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/reservations")
class ReservationController (
    private val reservationUseCase: ReservationUseCase
) {
    @PostMapping
    fun reserve(@RequestBody @Valid request: ReservationRequest) : List<ReservationResponse> {
        val reservations = reservationUseCase.reserve(request.userId, request.screeningId, request.seatIds)
        return reservations.map{ ReservationResponse.from(it) }
    }
}